Mathlib/Algebra/Module/ZLattice/Summable.lean

```diff
@@ -0,0 +1,285 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset.Disjoint
+import Mathlib.Algebra.Module.ZLattice.Basic
+import Mathlib.Analysis.PSeries
+
+/-!
+# Convergence of `p`-series on lattices
+
+Let `E` be a finite dimensional normed `ℝ`-space, and `L` a discrete subgroup of `E` of rank `d`.
+We show that `∑ z ∈ L, ‖z - x‖ʳ` is convergent for `r < -d`.
+
+## Main results
+- `ZLattice.summable_norm_rpow`: `∑ z ∈ L, ‖z‖ʳ` converges when `r < -d`.
+- `ZLattice.summable_norm_sub_rpow`: `∑ z ∈ L, ‖z - x‖ʳ` converges when `r < -d`.
+- `ZLattice.tsum_norm_rpow_le`:
+  `∑ z ∈ L, ‖z‖ʳ ≤ Aʳ * ∑ k : ℕ, kᵈ⁺ʳ⁻¹` for some `A > 0` depending only on `L`.
+
+-/
+
+noncomputable section
+
+open Module
+
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+variable [FiniteDimensional ℝ E] {L : Submodule ℤ E} [DiscreteTopology L]
+variable {ι : Type*} (b : Basis ι ℤ L)
+
+lemma ZLattice.exists_forall_abs_repr_le_norm :
+    ∃ (ε : ℝ), 0 < ε ∧ ∀ (x : L), ∀ i, ε * |b.repr x i| ≤ ‖x‖ := by
```

Comment:
Doesn't this generalise away from the reals?

Reply:
I can't find a meaningful generalization as we don't have $$K$$-valued norm etc.